src/snowflake/snowpark/dataframe_writer.py

```diff
@@ -89,18 +90,23 @@ def save_as_table(
 
                 "ignore": Ignore this operation if data already exists.
 
-            create_temp_table: The to-be-created table will be temporary if this is set to ``True``.
+            create_temp_table: (Deprecated) The to-be-created table will be temporary if this is set to ``True``.
+            table_type: The table type of to-be-created table. The supported table types include ``temp``/``temporary``
+                        and ``transient``. An empty string means permanent table as per SQL convention.
```

Comment:
Optionally, I suggest changing the description of table_type to something like this:

The table type of table to be created. The supported table types are: ``temp``, ``temporary``,
and ``transient``. An empty string means to create a permanent table.

Reply:
Do you think it might be helpful to include a link to documentation about temporary and transient tables? Here's an article that might be suitable: 

https://docs.snowflake.com/en/user-guide/tables-temp-transient.html

Reply:
Edited and added the link as well :)